Feature/rename frame to span

wysiwys opened the pull request:
This PR renames frame -> span in the code and documentation, where applicable (see issue #660 and discussion in #652). 

One question that is open is about some differing terminology in the minimp3 decoder. Specifically, the mp3  `Frame`s from the minimp3 crate are interpreted as spans in the new rodio terminology. Therefore, when renaming values used by this decoder, should the term "frame" or "span" be preferred where possible? 

This code snippet from `src/decoder/mp3.rs`  illustrates this question:

```rust
pub struct Mp3Decoder<R>
where
    R: Read + Seek,
{
    // decoder: SeekDecoder<R>,
    decoder: Decoder<R>,
    // minimp3 Frames are interpreted as rodio spans
    current_span: Frame,
    current_span_offset: usize,
}
```
I went with this approach for now, because it seems quite explicit about the interpretation of `Frame`s, but I can also see the argument for matching the struct member name to its type.

Thanks

dvdsk commented on src/decoder/mp3.rs:
```diff
@@ -14,8 +14,9 @@ where
     decoder: Decoder<R>,
-    current_frame: Frame,
-    current_frame_offset: usize,
+    // minimp3 Frames are interpreted as rodio spans
```
Maybe: `// What minimp3 calls frames rodio calls spans`

and then fully qualify the path to `Frame` (something like `minimp3::Frame`)?

dvdsk reviewed (changes requested):
Thanks a lot, must have been quite the chore to do all this!

Good catch on the minim3 Frame, rodio span. I agree a comment is the best solution there. I added a suggestion on how we can improve that. Can you also add an entry in the changelog? I think the only public API change is the name of the trait method. 

Thats all for the rest it looks perfect!

wysiwys commented:
No problem! I have made those changes and added a changelog entry for the public API change. 

Do these updates to the changelog look good? I added one other entry about the renaming of 'frame' to 'span' in the crate and documentation. However, I can also combine both new entries into one, if that would be better.

dvdsk commented:
> I added one other entry about the renaming of 'frame' to 'span' in the crate and documentation.

good idea! Only thing you missed is the `-` character for starting a new item :)  But I'll fix that post merge :+1:. LGTM :tada: 

@PetrGlad @iluvcapra you might run into some merge conflicts because of this change. It should be pretty straight forward to fix.

dvdsk commented:
> > 
> 
> good idea! Only thing you missed is the `-` character for starting a new item :) But I'll fix that post merge 👍. LGTM 🎉

nvmd everything was perfect already. I think the github diff webui ate the `-`
